drivers/storage/s3fs/storage/s3fs_storage.go

```diff
@@ -0,0 +1,392 @@
+			"error deleting s3 bucket")
+		return goof.WithFieldE(
+			"volumeID", volumeID, "error deleting s3 bucket", err)
+	}
+
+	return nil
+}
+
+// VolumeAttach attaches a volume and provides a token clients can use
+// to validate that device has appeared locally.
+func (d *driver) VolumeAttach(
+	ctx types.Context,
+	volumeID string,
+	opts *types.VolumeAttachOpts) (*types.Volume, string, error) {
+
+	vol, err := d.getVolume(ctx, volumeID, types.VolAttReq)
+	if err != nil {
+		return nil, "", goof.WithFieldE(
+			"volumeID", volumeID,
+			"volume is not in bucket list", err)
+	}
+	return vol, "", nil
+}
+
+// VolumeDetach detaches a volume.
+func (d *driver) VolumeDetach(
+	ctx types.Context,
+	volumeID string,
+	opts *types.VolumeDetachOpts) (*types.Volume, error) {
+
+	vol, err := d.getVolume(ctx, volumeID, types.VolAttReq)
+	if err != nil {
+		return nil, goof.WithFieldE(
+			"volumeID", volumeID,
+			"volume is not in bucket list", err)
+	}
+	return vol, nil
+}
+
+// Snapshots returns all volumes or a filtered list of snapshots.
+func (d *driver) Snapshots(
+	ctx types.Context,
+	opts types.Store) ([]*types.Snapshot, error) {
+	// TODO Snapshots are not implemented yet
+	return nil, types.ErrNotImplemented
+}
+
+// SnapshotInspect inspects a single snapshot.
+func (d *driver) SnapshotInspect(
+	ctx types.Context,
+	snapshotID string,
+	opts types.Store) (*types.Snapshot, error) {
+	// TODO Snapshots are not implemented yet
+	return nil, types.ErrNotImplemented
+}
+
+// SnapshotCopy copies an existing snapshot.
+func (d *driver) SnapshotCopy(
+	ctx types.Context,
+	snapshotID, snapshotName, destinationID string,
+	opts types.Store) (*types.Snapshot, error) {
+	// TODO Snapshots are not implemented yet
+	return nil, types.ErrNotImplemented
+}
+
+// SnapshotRemove removes a snapshot.
+func (d *driver) SnapshotRemove(
+	ctx types.Context,
+	snapshotID string,
+	opts types.Store) error {
+	// TODO Snapshots are not implemented yet
+	return types.ErrNotImplemented
+}
+
+var errGetLocDevs = goof.New("error getting local devices from context")
+
+func (d *driver) toTypeVolumes(
+	ctx types.Context,
+	attachments types.VolumeAttachmentsTypes) ([]*types.Volume, error) {
+
+	res, err := d.svc.ListBuckets(&awss3.ListBucketsInput{})
+	if err != nil {
+		ctx.WithError(err).Error("error listing s3 buckets")
+		return nil, goof.WithError("error listing s3 buckets", err)
+	}
+
+	var vols []*types.Volume
+	for _, b := range res.Buckets {
+		vols = append(vols, d.toTypeVolume(ctx, *b.Name, attachments))
+	}
+	return vols, nil
+}
+
+func (d *driver) toTypeVolume(
+	ctx types.Context,
+	bucket string,
+	attachments types.VolumeAttachmentsTypes) *types.Volume {
+
+	vol := &types.Volume{
+		Name: bucket,
+		ID:   bucket,
+	}
+
+	if attachments.Requested() {
+		vatt := &types.VolumeAttachment{
+			VolumeID:   bucket,
+			DeviceName: bucket,
+		}
+		if iid, iidOK := context.InstanceID(ctx); iidOK {
+			vatt.InstanceID = iid
```

Comment:
Hi @codenrhoden,

Ah, you're right. I should make it so all volumes are available if the instance ID is not provided. Thanks.